src/helpers/index.ts

```diff
@@ -6,22 +6,18 @@ export const getRegistrarForDomain = async (
   hub: ZNSHub,
   domainId: string
 ): Promise<Registrar> => {
-  try {
-    let registrarAddress;
-    try {
-      registrarAddress = await hub.domainToContract(domainId);
-    } catch (e) {
-      throw Error(`Failed to access hub: ${e}`);
-    }
+  let registrarAddress;
 
-    if (registrarAddress === ethers.constants.AddressZero) {
-      throw Error(`Null address for registrar.`);
-    }
-    const contract = getRegistrar(hub.provider, registrarAddress);
-    return contract;
+  try {
+    registrarAddress = await hub.getRegistrarForDomain(domainId)
   } catch (e) {
-    throw Error(
-      `Failed to get Registrar for domain '${domainId}' Reason: ${e}`
-    );
+    throw Error(`Failed to access hub: ${e}`);
   }
+
+  if (registrarAddress === ethers.constants.AddressZero) {
+    throw Error(`Null address for registrar.`);
```

Comment:
Error message should contain better context, "Hub returned null address when trying to find registrar for domain"